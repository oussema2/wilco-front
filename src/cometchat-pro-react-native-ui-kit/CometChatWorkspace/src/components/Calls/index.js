export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
