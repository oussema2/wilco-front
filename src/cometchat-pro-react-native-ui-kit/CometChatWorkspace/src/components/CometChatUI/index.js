/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
