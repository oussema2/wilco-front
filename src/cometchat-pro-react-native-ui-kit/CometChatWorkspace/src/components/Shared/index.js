export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
