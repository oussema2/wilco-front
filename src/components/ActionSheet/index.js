export { default as ActionSheet } from './ActionSheet';
