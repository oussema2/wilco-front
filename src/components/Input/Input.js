import React, { useState } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { Input as InputElement } from 'react-native-elements';
import { useFocus } from '../../hooks/useFocus';
import { PasswordVisibilitySwitch } from '../PasswordVisibilitySwitch';
import {
	defaultStyle,
	focusedStyle,
	errorStyle,
	disabledStyle,
	viewStyles,
	errorMessageHiddenStyle
} from './Input.styles';
import { fonts, palette } from '../../Theme';
import ClearText from './components/ClearText';
import { ActivityIndicator } from '../ActivityIndicator';

const Input = ( {
	testID,
	value,
	error,
	disabled,
	isPassword,
	onFocus,
	onBlur,
	onChange,
	hideError,
	placeholder,
	autoCapitalize,
	onSubmitEditing,
	returnKeyType,
	scrollEnabled,
	clearText,
	isLoading,
	helperText,
	autoCorrect
} ) => {
	const [ focused, onInputFocused, onInputBlurred ] = useFocus( onFocus, onBlur );
	const [ showPassword, setShowPassword ] = useState( false );

	let styles = defaultStyle;
	if ( focused ) { styles = { ...styles, ...focusedStyle }; }
	if ( error ) { styles = { ...styles, ...errorStyle }; }
	if ( disabled ) { styles = { ...styles, ...disabledStyle }; }

	let errorMessageStyles = styles.errorMessage;
	const helperTextStyle = error ? styles.helperText : styles.helperTextWithoutError;
	if ( hideError ) { errorMessageStyles = errorMessageHiddenStyle.errorMessage; }

	const _renderPasswordVisibilitySwitch = () => (
		<PasswordVisibilitySwitch
			testID="password-switch"
			passwordVisible={showPassword}
			onChange={() => setShowPassword( !showPassword )}
		/>
	);

	const _renderClearText = () => (
		<ClearText
			testID="clear-text-switch"
			value={value}
			onPress={clearText}
		/>
	);

	const _renderIsLoading = () => (
		<ActivityIndicator
			testID="activity-indicator"
			size="small"
		/>
	);

	const _setFontFamily = ( ref ) => ref && ref.props && ref.setNativeProps( {
		text: ref.props.value,
		style: ( ref.props.value ) ? fonts.bodyFocus : fonts.body // works only on iOS
	} );

	// eslint-disable-next-line consistent-return
	const _renderRightIcon = () => {
		if ( isPassword ) return _renderPasswordVisibilitySwitch();
		if ( isLoading ) return _renderIsLoading();
		if ( clearText ) return _renderClearText();
	};

	return (
		<View style={viewStyles.container}>
			<InputElement
				ref={_setFontFamily}
				testID={testID}
				containerStyle={styles.container}
				inputContainerStyle={styles.inputContainer}
				inputStyle={styles.input}
				selectionColor={palette.primary.default}
				placeholderTextColor={palette.grayscale.aluminum}
				value={value}
				disabled={disabled}
				onFocus={onInputFocused}
				onBlur={onInputBlurred}
				onChangeText={onChange}
				autoCapitalize={autoCapitalize}
				autoCorrect={autoCorrect}
				secureTextEntry={isPassword && !showPassword}
				textAlignVertical="center"
				errorMessage={error}
				errorStyle={errorMessageStyles}
				placeholder={placeholder}
				rightIcon={_renderRightIcon}
				onSubmitEditing={onSubmitEditing}
				returnKeyType={returnKeyType}
				scrollEnabled={scrollEnabled}
			/>
			{helperText && !error && (
				<Text testID="helperText-testID" style={helperTextStyle}>
					{helperText}
				</Text>
			)}
		</View>
	);
};

Input.propTypes = {
	testID: PropTypes.string,
	value: PropTypes.string,
	error: PropTypes.oneOfType( [ PropTypes.string, PropTypes.bool ] ),
	disabled: PropTypes.bool,
	isPassword: PropTypes.bool,
	onFocus: PropTypes.func,
	onBlur: PropTypes.func,
	onChange: PropTypes.func,
	hideError: PropTypes.bool,
	placeholder: PropTypes.string,
	autoCapitalize: PropTypes.oneOf( [ 'characters', 'words', 'sentences', 'none' ] ),
	onSubmitEditing: PropTypes.func,
	returnKeyType: PropTypes.oneOf( [ 'done', 'go', 'next', 'search', 'send', 'default' ] ),
	scrollEnabled: PropTypes.bool,
	clearText: PropTypes.func,
	isLoading: PropTypes.bool,
	helperText: PropTypes.string,
	autoCorrect: PropTypes.bool
};

Input.defaultProps = {
	testID: 'Input-Component',
	value: '',
	error: '',
	disabled: false,
	isPassword: false,
	onFocus: () => {},
	onBlur: () => {},
	onChange: () => {},
	hideError: false,
	placeholder: '',
	autoCapitalize: 'none',
	onSubmitEditing: () => {},
	returnKeyType: 'default',
	scrollEnabled: true,
	clearText: null,
	isLoading: null,
	helperText: null,
	autoCorrect: true
};

export default Input;
