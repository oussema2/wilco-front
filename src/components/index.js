export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
