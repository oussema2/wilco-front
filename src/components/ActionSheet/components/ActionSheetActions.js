import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
