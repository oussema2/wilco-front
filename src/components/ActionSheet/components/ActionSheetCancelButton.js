import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
