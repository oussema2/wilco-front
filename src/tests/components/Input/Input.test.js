import React from 'react';
import { render } from '@testing-library/react-native';
import { Input } from '../../../components/Input';

describe( '<Input />', () => {
	let component;
	const testID = 'testing-Input-Component';

	const setUp = ( props ) => {
		component = render( <Input testID={testID} {...props} /> );
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the Input component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();

			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with the isPassword', () => {
		it( 'renders the password visibility switch', () => {
			expect( component.queryByTestId( 'password-switch' ) ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with the helperText', () => {
		it( 'renders the helper text', () => {
			const helperText = 'Helper text';
			setUp( { helperText } );
			expect( component.queryByTestId( 'helperText-testID' ) ).toHaveTextContent( helperText );
		} );
	} );
} );
