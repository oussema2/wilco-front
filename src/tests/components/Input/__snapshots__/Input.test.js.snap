// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> with the default props renders the Input component correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F2F2F2",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "borderRadius": 8,
          "flexDirection": "row",
          "marginBottom": 5,
          "minHeight": 52,
          "paddingHorizontal": 12,
          "textAlignVertical": "center",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder=""
        placeholderTextColor="#A0A0A0"
        returnKeyType="default"
        scrollEnabled={true}
        secureTextEntry={false}
        selectionColor="#12A3BE"
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "minHeight": 40,
            "paddingVertical": 0,
          }
        }
        testID="testing-Input-Component"
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#D60A50",
          "fontFamily": "NotoSansJP-Regular",
          "fontSize": 12,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 16,
          "margin": 0,
          "marginLeft": 0,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`<Input /> with the isPassword renders the password visibility switch 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F2F2F2",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "borderRadius": 8,
          "flexDirection": "row",
          "marginBottom": 5,
          "minHeight": 52,
          "paddingHorizontal": 12,
          "textAlignVertical": "center",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder=""
        placeholderTextColor="#A0A0A0"
        returnKeyType="default"
        scrollEnabled={true}
        secureTextEntry={false}
        selectionColor="#12A3BE"
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "minHeight": 40,
            "paddingVertical": 0,
          }
        }
        testID="testing-Input-Component"
        textAlignVertical="center"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#D60A50",
          "fontFamily": "NotoSansJP-Regular",
          "fontSize": 12,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 16,
          "margin": 0,
          "marginLeft": 0,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
